src/viztracer/modules/vcompressor/vc_dump.c

```diff
@@ -258,3 +267,160 @@ load_events_from_file(FILE* fptr)
     return parsed_events;
 
 }
+
+
+int dump_file_info(PyObject* file_info, FILE* fptr)
+{
+    const char * buffer = NULL;
+    uint64_t content_length = 0;
+    uint64_t compression_length = 0;
+    PyObject* json_ret = NULL;    
+    PyObject* zlib_ret = NULL;
+    PyObject* bytes_data = NULL;
+    PyObject* dumps_func = NULL;
+    PyObject* compress_func = NULL;
+    PyObject* json_args = NULL;
+    PyObject* zlib_args = NULL;
+    
+    dumps_func = PyObject_GetAttrString(json_module, "dumps");
+    if (!dumps_func) {
+        goto clean_exit;
+    }
+
+    compress_func = PyObject_GetAttrString(zlib_module, "compress");
+    if (!compress_func) {
+        goto clean_exit;
+    }
+
+    // json dumps file_info
+    json_args = PyTuple_New(1);
+    // file_info points to the file_info in json object we compress
+    // PyTuple_SetItem steals the reference, but we don't want to release file_info here
+    // So we need to call Py_INCREF here
+    PyTuple_SetItem(json_args, 0, file_info);
+    Py_INCREF(file_info); 
+    json_ret = PyObject_CallObject(dumps_func, json_args);
+    Py_DECREF(json_args);
+    if (!json_ret){
+        goto clean_exit;
+    }
+
+    // convert string to bytes
+    bytes_data = PyObject_CallMethod(json_ret, "encode", NULL);
```

Comment:
Also, I know it's a long shot, but we should probably also check `bytes_data` is a `PyByte`. We assumed `PyBytes_Size` will succeed and that relies on the fact that `bytes_data` is a `PyByte`. I know `encode` should always return a byte on success, but this could cause a segfault when say in the future, encode somehow returns something else.

Let's do a `if (!PyBytes_Check(bytes_data))` for error handling just to be safe. I'm okay if my code raises exceptions on something I do not fully cover, but segfault is the thing we should try to avoid.

Reply:
> Seems like `bytes_data` is never released.

I didn't call `Py_DECREF(bytes_data);` because `zlib_args` steals the reference of `bytes_data`. `Py_DECREF(zlib_args);` will release it.

Reply:
> > Seems like `bytes_data` is never released.
> 
> I didn't call `Py_DECREF(bytes_data);` because `zlib_args` steals the reference of `bytes_data`. `Py_DECREF(zlib_args);` will release it.

Right. Tuple is tricky.